### PR TITLE
Fix to disallow special char and < 3 char searches

### PR DESCRIPTION
**Pull request by pumazi**
It's not exactly pretty, but it should work. The search view is a mess, so adding this condition to that mess isn't going to make it that much worse. 😛

### CODE REVIEW

**Comment by pumazi** [comment]
Since this PR is a fork from `v2.5.1`, we'll need to ignore some issues that have been corrected since then. The failing tests (at least on the push) are due to a change in the encoding procedure in webob a dependency of pyramid.

File: cnxarchive/tests/test_views.py [hunk 1]
```diff
@@ -2103,8 +2103,63 @@ def test_search_whitespace(self):
...
+            u'results': {
+                u'items': [],
+                u'total': 0,
+                u'limits': [],
+                },
+            }))
+
     def test_search_utf8(self):
-        self.request.params = {'q': '"你好"'}
+        self.request.params = {'q': '"今天你好吗"'}
```
Comment (reedstrm) [inline cnxarchive/tests/test_views.py]: Hmm, I bet the 2-char Chinese search is sufficiently meaningful that we shouldn't disallow it. I'd bet single-char are even useful.

File: cnxarchive/search.py [hunk 2]
```diff
@@ -37,6 +37,7 @@
                  [chr(i) for i in range(ord('a'), ord('z') + 1)])
 WILDCARD_KEYWORD = 'text'
 VALID_FILTER_KEYWORDS = ('type', 'pubYear', 'authorID', 'submitterID')
+SEARCH_SPECIAL_CHARS = '"\',:'
```
Comment (reedstrm) [inline cnxarchive/search.py]: Hmm, currently we can search for contractions like "don't" which can actually be useful.

Comment (pumazi) [inline cnxarchive/search.py]: This won't affect how searches are done. Stripped, `don't` will be `dont` which is a valid search.

File: cnxarchive/tests/test_views.py (hunk 1 above)
Comment (pumazi) [inline cnxarchive/tests/test_views.py]: I get what you are saying here, but I don't think there is a simple fix.

**Comment by reedstrm** [comment]
Perhaps make this a disallow the completely blank query, only?

File: cnxarchive/search.py (hunk 2 above)
Comment (reedstrm) [inline cnxarchive/search.py]: Ah right, strip() only removes from the extremes of the string. so "" and '' and :: and ,, would all string to nothing. How about we keep that, but change the numeric limit to < 2, or remove it completely? I'm particularly concerned about single-char Chinese being useful searches. Two char are definitely useful

Comment (reedstrm) [inline cnxarchive/search.py]: Should we add `space` to the SEARCH_SPECIAL_CHARS, so that `" "` collapses as  well?

**Pull request closed without merging**